client/client.go

```diff
@@ -46,14 +49,23 @@ func NewClient(baseUrl, clientId, clientSecret string) (*Client, error) {
 	if err != nil {
 		return nil, err
 	}
+
 	return &Client{
-		HTTPClient: &http.Client{Timeout: 10 * time.Second},
-		BaseURL:    baseUrl,
-		Token:      credentials.AccessToken,
+		HTTPClient: &http.Client{
+			Timeout: 30 * time.Second,
+		},
+		BaseURL:     baseUrl,
+		Token:       credentials.AccessToken,
+		RateLimiter: rate.NewLimiter(rate.Every(1*time.Second), 5),
 	}, nil
 }
 
 func (c *Client) DoRequest(req *http.Request) ([]byte, error) {
+	err := c.RateLimiter.Wait(context.Background())
```

Comment:
Finally! We will have rate limiting. Thanks. :)
It will probably fix #11.

Reply:
hopefully 🤞